2103 contact tracing various

clemens-msupply opened the pull request:
<!-- IMPORTANT!
  - Every PR must reference an issue; this helps to explain the intent of the PR
 -->

Part of #2103 

- Rename contact trace tab to "Contact Tracing"
- Removes contact tracing status (needs more discussion and is not needed right now)
- Remove status from UI
- Add first and last name
- Add dob and gender
- Remove full contact tracing list view
- Add store_id for each contact trace

# Testing:
- Check contact tracing detail view and list view
- Check that a contact trace DB entry has the current store id

mark-prins commented on server/service/src/programs/schemas/contact_trace.json:
```diff
@@ -100,17 +100,27 @@
-        "status": {
-          "$ref": "#/definitions/ContactTraceStatus",
-          "description": "Current status"
+        "location": {
```
I couldn't save the contacts - and then noticed that some new bits were missing. I updated using this JSON schema but then had issues because my ui_schema was out of date.

so, I took the ui_schema from the programschemas repo, which caused more problems.

I've now taken the JSON from that repo as well, and things are working. But the JSON differs to this file! Does this one require updating?

clemens-msupply replied:
Hmm its working here and copied over the latest version and there is no diff. Maybe run `./build_all.sh` to generate the latest json schemas...

mark-prins commented on client/packages/system/src/Patient/ContactTrace/Components/columns.ts:
```diff
@@ -11,55 +11,42 @@ import { ContactTraceRowFragment } from '@openmsupply-client/programs';
-      key: 'contactTraceId',
-      label: 'label.contact_trace-id',
+      key: 'datetime',
+      label: 'label.date',
```
the label in the issue is `Created/Added on` (which I think is better as `Date created` or `Created date` anyway)

mark-prins commented on client/packages/system/src/Patient/ContactTrace/Components/columns.ts:
```diff
@@ -11,55 +11,42 @@ import { ContactTraceRowFragment } from '@openmsupply-client/programs';
-        dateString ? localisedDate((dateString as string) || '') : '',
-    }
-  );
+    },
```
The issue asked for `Gender` and `Contacted?` as well

clemens-msupply replied:
I added Gender but @richardmoizeau and I decited to leave "Contacted" out... (Planned to put it into the general contact tracing status but this would need some more thought...)

mark-prins commented on client/packages/system/src/Patient/ContactTrace/Components/columns.ts:
```diff
@@ -11,55 +11,42 @@ import { ContactTraceRowFragment } from '@openmsupply-client/programs';
     {
-      key: 'datetime',
-      label: 'label.date',
+      key: 'age',
```
this one needs `sortable: false` added

mark-prins replied:
alternatively, you could:

```
    {
      key: 'dateOfBirth',
      label: 'label.age',
      align: ColumnAlign.Right,
      width: 175,
      accessor: ({ rowData }) => rowData.age,
    },
```

clemens-msupply replied:
Thanks did this

mark-prins commented on server/repository/src/db_diesel/contact_trace.rs:
```diff
@@ -142,6 +146,12 @@ impl<'a> ContactTraceRepository<'a> {
                 ContactTraceSortField::LastName => {
                     apply_sort!(query, sort, contact_trace_dsl::last_name)
                 }
+                ContactTraceSortField::Gender => {
```
these will need to be added to the graphql queries `ContactTraceSortInput` too

mark-prins reviewed (changes requested):
Looking great! I have a couple of things to tweak, as noted. And an apology!

Apologies - I've snuck in an unrelated import fix in 886188352
The incorrect import came in with prescriptions and I thought it had been fixed already 🤷 

Should the label be `Sex` (current) or `Gender` (in issue)?

clemens-msupply commented on server/repository/src/migrations/v1_03_00/contact_trace.rs:
```diff
@@ -11,34 +11,33 @@ pub(crate) fn migrate(connection: &StorageConnection) -> anyhow::Result<()> {
           first_name TEXT,
-          last_name TEXT
+          last_name TEXT,
+          gender TEXT,
```
Yes I have to move this out into another migration 1.3.1 or 1.4.0?

clemens-msupply replied:
I moved it to 1.4.0 for now

mark-prins replied:
cool - thanks. I would have said 1.3.1 but the consensus is to release a full minor version, so you are correct. 1.4.0 it is.

clemens-msupply commented on client/packages/system/src/ContactTrace/DetailView/useContactTraceData.tsx:
```diff
@@ -1,16 +1,18 @@
   useDocumentRegistry,
 } from '@openmsupply-client/programs';
 import { usePatient } from '../../Patient';
+import { useAuthContext } from 'packages/common/src';
```
That would be great!

clemens-msupply commented:
Thanks I think I addressed all comments

> Should the label be `Sex` (current) or `Gender` (in issue)?

Client uses Sex, e.g. in custom forms but our views use Gender. Would be cool to be able to configure it though...

mark-prins reviewed (approved):
👍 ( sorry, took me a while to get back and retest! )
